numpyro/distributions/distribution.py

```diff
@@ -269,6 +269,13 @@ def _validate_sample(self, value):
 
     def __call__(self, *args, **kwargs):
         key = kwargs.pop('rng_key')
+        # FIXME: This also raises error for Delta site.
+        # Is there a better way to raise this error?
```

Comment:
I get a better way is to add an assertation at `sample` method of all distributions that requires `sample`. It is a bit verbose but will resolve the following situations:
+ Delta site does not need `rng_key`
+ The error should happen at the base_dist of ExpandedDistribution, rather ExpandedDistribution itself. Current, when we missed `rng_key` at `ExpandedDistribution`, we got the following error: `ValueError: Missing random key for ExpandedDistribution distribution.`

Reply:
What do you think, @neerajprad?

Reply:
I agree that it will be better to add it in `sample`. That way, if we use the distributions directly, it will still raise an error if the rng key is absent.

Reply:
That is a good point. I'll make a change.

Reply:
Done. :)